docs/infrastructure/README.md

```diff
@@ -0,0 +1,167 @@
+# Dotkom Container Service (Evergreen)
+
+Evergreen is our container service that should be used to run all Monoweb applications in all environments, including
+production and staging. Evergreen is based on Docker containers, and we use Docker Compose as the orchestrator.
+
+At its core, Evergreen is a single Traefik load balancer, and a set of Docker containers running the various apps that
+are part of Monoweb. To keep simplicity low, Evergreen is hosted on a single AWS EC2 instance. Because of this, the
+services are not intended to be highly scalable, and should likely be used for smaller applications that we don't mind
+small downtimes for.
+
+**Table of Contents**
+
+- [Motivation](#motivation)
+- [Service Architecture](#service-architecture)
+- [Deployment & Scaling](#deployment--scaling)
+- [Pricing](#price)
+- [Pitfalls](#pitfalls--alternatives)
+
+## Motivation
+
+The motivation for Evergreen is to provide a really cheap way to run multiple small applications that can tolerate
+occasional downtime. Additionally, we want to use Docker as the deployment format. Furthermore, Docker Compose is
+relatively simple, and we want to keep deployment complexity low in order to make it easy for new developers to
+understand.
+
+The proposed solution has a set of pitfalls outlined in the [Pitfalls & Alternatives](#pitfalls--alternatives) section,
+but we believe that keeping things simple and cheap is rather important for a student organization with limited
+financial resources.
+
+## Service Architecture
+
+The entire application fleet is hosted on a single AWS EC2 (t4g.medium as of writing) instance. The instance is
+available to the public internet through a VPC, but all traffic should be routed from CloudFront to Traefik. Traefik is
+responsible for load balancing and routing across the various applications.
+
+Each application is packaged as a Docker container that is pushed to AWS ECR. GitHub Actions dispatches from push or
+tag release events, and pulls the latest Docker image from ECR, and restarts the relevant containers.
+
+Environment variables are pulled from Doppler and injected into the container through a .env file stored on the
+Evergreen instance.
+
+To allow fair share of the instance resources, each container is limited to a subset of the vCPU and memory available on
+the instance. This is configured in the compose configuration file.
+
+The EC2 instance is an Amazon Linux 2023 AMI, and is built with Packer through the terraform monorepo.
+
+In order to allow for instances to be re-deployed or scaled, the Evergreen service is configured behind CloudFront. Each
+time a new Evergreen instance is deployed, the routing table for CloudFront is updated to point to the new instance,
+and the entire CloudFront distribution is invalidated.
```

Comment:
`docker compose pull && docker compose up --force-recreate` would also work? Seems a bit much to rebuild the entire machine image each deploy if I understand it correctly?

(The document later talks about SSH-ing to do something like this?)

Reply:
Nothing builds on the ec2 instance, that all happens in gh actions

Reply:
Updated wording